processor/cumulativetodeltaprocessor/processor.go

```diff
@@ -16,35 +16,99 @@ package cumulativetodeltaprocessor
 
 import (
 	"context"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
+
+	awsmetrics "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/metrics"
 )
 
 type cumulativeToDeltaProcessor struct {
-	metrics []string
-	logger  *zap.Logger
+	metrics         []string
+	logger          *zap.Logger
+	deltaCalculator awsmetrics.MetricCalculator
 }
 
 func newCumulativeToDeltaProcessor(config *Config, logger *zap.Logger) *cumulativeToDeltaProcessor {
 	return &cumulativeToDeltaProcessor{
-		metrics: config.Metrics,
-		logger:  logger,
+		metrics:         config.Metrics,
+		logger:          logger,
+		deltaCalculator: newDeltaCalculator(),
 	}
 }
 
 // Start is invoked during service startup.
-func (mgp *cumulativeToDeltaProcessor) Start(context.Context, component.Host) error {
+func (ctdp *cumulativeToDeltaProcessor) Start(context.Context, component.Host) error {
 	return nil
 }
 
 // processMetrics implements the ProcessMetricsFunc type.
-func (mgp *cumulativeToDeltaProcessor) processMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+func (ctdp *cumulativeToDeltaProcessor) processMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+	inputMetricSet := make(map[string]bool)
+	for _, name := range ctdp.metrics {
+		inputMetricSet[name] = true
+	}
+
+	resourceMetricsSlice := md.ResourceMetrics()
+
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		rm := resourceMetricsSlice.At(i)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for i := 0; i < ilms.Len(); i++ {
+			ilm := ilms.At(i)
+			metricSlice := ilm.Metrics()
+			for j := 0; j < metricSlice.Len(); j++ {
+				metric := metricSlice.At(j)
+				_, ok := inputMetricSet[metric.Name()]
+				if ok {
```

Comment:
```suggestion
				if inputMetricSet[metric.Name()] {
```

this is a `map[string]bool` that doesn't appear to be modified at any point after it is created, a direct check for a `true` value works as well here and is clearer.

Reply:
Updated.